### PR TITLE
Migrate combobox to mwc

### PR DESCRIPTION
**Pull request by bramkragten**
## Proposed change

 Migrates combo box to mwc

image

Next step, migrate all comboboxes to use `ha-combobox`

## Type of change

- [X] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [X] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [ ] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: src/components/ha-combo-box.ts [hunk 1]
```diff
@@ -1,25 +1,53 @@
...
+      width: 100%;
+    }
+    [part="checkmark"] {
+      display: none;
+    }
+  `
+);
+
+const defaultRowRenderer: ComboBoxLitRenderer<string> = (item) =>
+  html` <mwc-list-item>${item}</mwc-list-item>`;
```
Comment (zsarnett) [inline src/components/ha-combo-box.ts:50]:
```suggestion
  html`<mwc-list-item>${item}</mwc-list-item>`;
```

File: src/components/ha-area-picker.ts [hunk 2]
```diff
@@ -3,7 +3,7 @@ import "@polymer/paper-input/paper-input";
 import "@polymer/paper-item/paper-item";
 import "@polymer/paper-item/paper-item-body";
 import "@polymer/paper-listbox/paper-listbox";
-import "@vaadin/vaadin-combo-box/theme/material/vaadin-combo-box-light";
+import "@vaadin/combo-box/theme/material/vaadin-combo-box-light";
```
Comment (balloob) [inline src/components/ha-area-picker.ts:6]: When no areas, this one looks weird:

image

File: src/components/entity/ha-entity-picker.ts [hunk 3]
```diff
@@ -2,7 +2,7 @@ import { mdiCheck, mdiClose, mdiMenuDown, mdiMenuUp } from "@mdi/js";
 import "@polymer/paper-input/paper-input";
 import "@polymer/paper-item/paper-icon-item";
 import "@polymer/paper-item/paper-item-body";
-import "@vaadin/vaadin-combo-box/theme/material/vaadin-combo-box-light";
+import "@vaadin/combo-box/theme/material/vaadin-combo-box-light";
```
Comment (balloob) [inline src/components/entity/ha-entity-picker.ts:5]: In target selector when no entities (not very common use case I know)

image

File: src/components/ha-area-picker.ts (hunk 2 above)
Comment (balloob) [reply src/components/ha-area-picker.ts:6]: Old screenshots from https://design.home-assistant.io/#automation/selectors

image

image

image

**Review by balloob — APPROVED** [review]
A few empty combobox issues left, rest looks good 👍

**Comment by bramkragten** [comment]
I just converted the base element now, not all comboboxes use the base element yet, that is the next step and should make all comboboxes look the same and good :-)

**Comment by steverep** [comment]
Fantastic.  Thanks for jumping right on this @bramkragten et al.

**Pull request merged**